### ISSUE TITLE
Right-click menu often disappears in Firefox right after appearing.

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#1728]
See below, this is in Firefox on a Mac...

![right-click](https://cloud.githubusercontent.com/assets/25212/22847884/fd6392ee-efa4-11e6-82ae-0dd78c82b5c8.gif)

Happens quite often (thought I can't tell exactly why) using both a mouse and right-clicking with the trackpad.

**Comment by Abykin** [comment]
Hey @flukeout I'd like to try to tackle this one.

**Comment by humphd** [comment]
@Abykin it's yours, go for it.

**Comment by OutsourcedGuru** [comment]
For what it's worth, Apple didn't have a "right" mouse button for years.  It was a single button.  The new Apple Magic Mouse out-of-box still doesn't have this native functionality but it can be added via System Preferences -> Mouse -> Enable Secondary Click.  And then you can actually click on the right side of the mouse and it will register differently than clicking straight down (noting that the entire body of the mouse moves down in both cases).  Alternatively, you can hold down the Control button while just clicking.

![imagic_mouse](https://cloud.githubusercontent.com/assets/15971213/23284781/36764e9a-f9e2-11e6-9db4-cd04d5a2c450.jpg)

And yet, you can still do the two-finger-tap on a Magic Mouse.

![magic_mouse-double-tap](https://cloud.githubusercontent.com/assets/15971213/23284870/b2e4ee14-f9e2-11e6-80e4-718b48d450b7.png)

By default on the trackpad, say, on a MacBook the contextual (in Windows we think of this as "right-mouse click") menu is opened via the two-finger tap mechanism.  It looks the same as the single-tap-with-two-fingers on the Magic Mouse.  (But don't confuse this with double-tapping-with-two-fingers, which is a different multi-gesture.)  As before, you can hold down the Control key and single-tap the trackpad.

When testing something like this, it would be good to work through all the various devices (trackpad, Magic Mouse and even a third-party multi-button mouse) to confirm a holistic fix rather than just a partial fix.

**Comment by flukeout** [comment]
Will definitely try this out on a few devices once we get a fix in. I'm getting the same behaviour with trackpad and right-clicking with a mouse right now.

Ideally, we'd have a way to access the menu by clicking an icon, as we've heard that younger users are unfamiliar with right-clicking.

**Comment by jgmac1106** [comment]
I think just having a copy and paste command on right click (if that is possible) woudl help. Right click just behaves in a way novice users have never expected and they probably do not know what three commands do.

**Comment by gideonthomas** [comment]
@jgmac1106 discussion is ongoing here: https://github.com/mozilla/brackets/issues/451, we're still trying to figure out how to make that possible.

**Comment by jgmac1106** [comment]
The same thing happens in gDocs but they provide a pop up modal when you
try to use right click and it tells you to use short cut keys to copy and
paste. Will leave comment in other issue

**Comment by humphd** [comment]
@flukeout can you reproduce this on https://bramble.mofostaging.net?  I can't.  I wonder if it got fixed when we merged with upstream.

**Comment by flukeout** [comment]
Here it is in staging.

![modal-pop](https://cloud.githubusercontent.com/assets/25212/23371776/1d0e61a6-fcce-11e6-86ea-0a6a0848fd48.gif)

**Comment by humphd** [comment]
@flukeout fascinating, I can't make it happen at all.  I've got Magic Mouse on Mac running FF dev edition.  What's the difference here?

**Comment by flukeout** [comment]
It also happens for me when my microsoft mouse is unplugged and I use my mac's trackpad, so ... I'm not sure.

**Comment by OutsourcedGuru** [comment]
@humphd Are you doing the two-finger-single-click action or did you turn on `System Preferences -> Mouse -> Enable Secondary Click` and you're doing a physical lean-on-the-right-side-of-the-top-of-the-Magic-Mouse right mouse click?

The latter does a pretty good job of simulating a standard Microsoft mouse in that the mousedown event lasts as long as you'd expect, at least in theory.  The former (or the Ctrl+click) doesn't sustain a mousedown event for very long.

**Comment by humphd** [comment]
I've tried both methods you describe, I can't reproduce it.  It's interesting that it's so different.

**Comment by OutsourcedGuru** [comment]
@humphd Wonder if it's fixed in your particular version/fork.

**Comment by humphd** [comment]
I'm testing on staging, same as Luke.

**Comment by OutsourcedGuru** [comment]
@humphd Would be interesting to know if you can right-click on Google Maps in FF as described on [this thread](https://productforums.google.com/forum/#!topic/maps/uuHPxBAMKBg).  Later in the thread, it was suggested that this bug was introduced in FF3.0, for what it's worth.  It's an interesting read, anyway.

**Comment by humphd** [comment]
Yeah google maps is fine for me.  fwiw, I'm using FF 53.0a2 (2017-02-13) (64-bit) on OS X 10.11.6.

**Comment by OutsourcedGuru** [comment]
Kind of a novel, but [this page](http://unixpapa.com/js/mouse.html) from 2011 seems to be suggesting that there are no click events for right button clicks in any browser.  See the "Event Triggering - Single Clicks" section.  Don't know if this is pertinent or not.  Right-mouse click is definitely in a gray area.

**Comment by OutsourcedGuru** [comment]
@humphd I was thinking more like... the BuildVersion out of /System/Library/Extensions/AppleHIDMouse.kext/Contents/verssion.plist

Maybe you and Luke are running different ones.

**Comment by humphd** [comment]
fwiw, mine is
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>BuildVersion</key>
	<string>3541</string>
	<key>CFBundleShortVersionString</key>
	<string>185.3</string>
	<key>CFBundleVersion</key>
	<string>185.3</string>
	<key>ProjectName</key>
	<string>AppleHIDMouse</string>
	<key>SourceVersion</key>
	<string>185003000000000</string>
</dict>
</plist>
```

**Comment by Abykin** [comment]
I've also had no success replicating this. Two-finger click and control+click both work fine.

**Comment by GrayHatter** [comment]
Can duplicate somewhat reliably on *nix Firefox proper  52.0.2 (and this is the only current result that comes up on google). So while it may not be the same issue, I'd like to know how/if you're able to fix this one.

**Comment by maxhebditch** [comment]
I'm finding that it only happens on highlighted text, and will stay open if no text is highlighted
![bug](https://user-images.githubusercontent.com/4867412/39424001-8cfe4de8-4c6c-11e8-822c-241b18a258b6.gif)
